packages/driver/cypress/integration/commands/actions/type_spec.js

```diff
@@ -557,6 +557,55 @@ describe('src/cy/commands/actions/type - #type', () => {
     })
   })
 
+  // https://github.com/cypress-io/cypress/issues/19541
+  describe(`type('{enter}') and click event on button-like elements`, () => {
+    beforeEach(() => {
+      cy.visit('fixtures/type-enter.html')
+    })
+
+    describe('triggers', () => {
+      const targets = [
+        'button-tag',
+        'input-button',
+        'input-image',
+        'input-reset',
+        'input-submit',
+      ]
+
+      targets.forEach((targetId) => {
+        it(`${targetId}`, () => {
+          cy.get(`#target-${targetId}`).click()
```

Comment:
If you use `cy.focus()` instead of `cy.click()`, then we don't have the extraneous 'click' event first.

```
          cy.get(`#target-${targetId}`).focus()
          cy.get(`#target-${targetId}`).type('{enter}')

          cy.get('li').eq(0).should('have.text', 'keydown')
          cy.get('li').eq(1).should('have.text', 'keypress')
          cy.get('li').eq(2).should('have.text', 'click')
          cy.get('li').eq(3).should('have.text', 'keyup')
```

Reply:
Fixed.